[TensorExpr] Delet `DimArg` class.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72390
* #72389

This class didn't add much value and only caused more boilerplate code.
This change removes the class and updates all the use cases with
uses of `ExprHandle`.

A side effect of this change is different names in loop variables, which
caused massive mechanical changes in our tests.

Differential Revision: [D34030296](https://our.internmc.facebook.com/intern/diff/D34030296)

Commit:
[TensorExpr] Delet `DimArg` class.

This class didn't add much value and only caused more boilerplate code.
This change removes the class and updates all the use cases with
uses of `ExprHandle`.

A side effect of this change is different names in loop variables, which
caused massive mechanical changes in our tests.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).

navahgar commented on torch/csrc/jit/tensorexpr/expr.cpp:
```diff
@@ -430,13 +430,13 @@ BufHandle Buf::make(const std::vector<ExprHandle>& dims, Dtype dtype) {
+    c10::optional<ExprHandle> initializer,
+    c10::optional<std::vector<ExprHandle>> strides,
+    c10::optional<ExprHandle> qscale,
+    c10::optional<ExprHandle> qzero) {
```
This was added in the previous PR, right? Wondering why clang-format didn't catch it there..

ZolotukhinM replied:
Hm, yeah, maybe the previous PR breaks clang-format :)

navahgar commented on torch/csrc/jit/tensorexpr/kernel.h:
```diff
@@ -198,7 +198,6 @@ class TORCH_API TensorExprKernel {
   void genInputDebugNames();
   void runKernel(Stack& stack);
 
-  std::vector<DimArg> dimsFromSizes(const std::vector<ExprHandle>& sizes);
```
This function wasn't defined anywhere!!! WTH :)

navahgar reviewed (approved):
LGTM

I didn't go through all the changes in tests, since they were mechanical.

Commit:
clang-format on "[TensorExpr] Delet `DimArg` class."


This class didn't add much value and only caused more boilerplate code.
This change removes the class and updates all the use cases with
uses of `ExprHandle`.

A side effect of this change is different names in loop variables, which
caused massive mechanical changes in our tests.

Differential Revision: [D34030296](https://our.internmc.facebook.com/intern/diff/D34030296)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).

Commit:
Update on "[TensorExpr] Delet `DimArg` class."


This class didn't add much value and only caused more boilerplate code.
This change removes the class and updates all the use cases with
uses of `ExprHandle`.

A side effect of this change is different names in loop variables, which
caused massive mechanical changes in our tests.

Differential Revision: [D34030296](https://our.internmc.facebook.com/intern/diff/D34030296)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).

Commit:
fix cuda test on "[TensorExpr] Delet `DimArg` class."


This class didn't add much value and only caused more boilerplate code.
This change removes the class and updates all the use cases with
uses of `ExprHandle`.

A side effect of this change is different names in loop variables, which
caused massive mechanical changes in our tests.

Differential Revision: [D34030296](https://our.internmc.facebook.com/intern/diff/D34030296)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34030296).
